### DISCUSSION TITLE
I cannot seem to make a tab active when the link is complex

### DISCUSSION THREAD

**Discussion by kevinr432** [sqlpage/SQLPage#679] [General]
I am building an application that will have some fi
![ScreenShot1](https://github.com/user-attachments/assets/1d697216-0334-412f-af4d-dd62cf0ec9a6)
![ScreenShot2](https://github.com/user-attachments/assets/b3f40dd8-7cb8-4a89-a271-d23ce0dbf5ee)
xed information in the upper portion of the screen.  I am using a tab control so the user can choose what additional information he wants to see in the lower portion of the screen.   It is working great.    But I wanted to make a tab ACTIVE to show the user which tab he selected.     I cannot seem to make a tab active when the link for the tab is complex.  Does anyone have any ideas.   See my two attached screen shots.    Please look at them in order #1, then #2.

**Comment by kevinr432** [comment]
Above in the red bubble where I said "But I cannot make the tab active when the index is complex", I meant "link" instead of "index"

**Comment by lovasoa** [comment]
Hello!

It seems you are connected to a Microsoft SQL Server database, which does not have a dedicated boolean type or true/false keywords. Instead, it represents boolean values using 1 (true) and 0 (false).

The reason why your initial example works is that SQLPage can interpret simple, static SQL queries internally without needing to send them to the SQL Server database. However, when your query is more complex, SQLPage forwards it to the SQL Server, where the difference in handling true/false values (instead of 1 and 0) causes the issue.


The solution is to use `1 as active` instead of `true as active`.

**Comment by kevinr432** [comment]
I changed from True/False to 1/0 and it works great now.  Thank you !!!!!

**Comment by lovasoa** [comment]
We fixed the sql parser (https://github.com/apache/datafusion-sqlparser-rs/pull/1510).
In the next version of sqlpage, your first example will give the same error as the second one, which should be less confusing.
